src/main/perl/LV.pm

```diff
@@ -59,7 +67,19 @@ sub _initialize
     $self->{volume_group} = NCM::LVM->new(BASEPATH . VGS . $st->{volume_group}, $config);
     $self->{size}         = $st->{size};
     $self->{stripe_size}  = $st->{stripe_size} if exists $st->{stripe_size};
-    
+    if (exists $st->{cache}) {
+        $self->{cache} = $st->{cache};
+        $self->{cache_lv} = NCM::LV->new (BASEPATH . LVS . $self->{cache}->{cache_lv}, $config);
+    }
+    if (exists $st->{devices}) {
+        $self->{devices} = [];
+        foreach my $devpath (@{$st->{devices}}) {
+            my $dev = NCM::BlockdevFactory::build($config, $devpath);
```

Comment:
can you check if `build` always returns something? e.g. in case of error/undef, this should be handled here too